src/awkward/_v2/contents/numpyarray.py

```diff
@@ -691,6 +691,22 @@ def _validityerror(self, path):
                 )
         return ""
 
+    def _rpad(self, target, axis, depth, clip):
+        if len(self.shape) == 0:
+            raise ValueError("cannot rpad a scalar")
+        elif len(self.shape) > 1 or not self.iscontiguous():
+            return self.toRegularArray()._rpad(target, axis, depth, clip)
+        posaxis = self.axis_wrap_if_negative(axis)
+        if posaxis != depth:
+            raise ValueError("axis exceeds the depth of this array")
```

Comment:
Here's another `np.AxisError`. Please search for all of these in your PR; I'm not necessarily going to find them all. Thanks!

Reply:
I caught all the bad axis errors and I am raising `np.AxisError` for them.